File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        unordered_map<int,int> freq;
        for(auto i:nums){
            freq[i]++;
        }
        int a=n/2;
        for(auto x: freq){
            if(x.second>a){
                return x.first;
            }
        }
        return 0;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
public int majorityElement(int[]nums) {
        int n=nums.length;
        int ans=0;
        HashMap<Integer, Integer>freq=new HashMap<>();
        for(int i:nums){
            freq.put(i,freq.getOrDefault(i,0)+1);
        }
        for(Map.Entry<Integer,Integer> i:freq.entrySet()){
            if( i.getValue()>n/2){
                ans=i.getKey();
            }
        }
        return ans;
    }
}